packages/beacon-node/src/chain/bls/jobItem.ts

```diff
@@ -73,12 +74,31 @@ export function jobItemWorkReq(job: JobQueueItem, format: PointFormat, metrics:
       const signatures = job.sets.map((set) => bls.Signature.fromBytes(set.signature, CoordType.affine, true));
       timer?.();
 
+      // adding verification randomness is swig specific. must not attempt with herumi until
+      // @chainsafe/bls is updated to support it with herumi
+      if (job.opts.addVerificationRandomness) {
+        const randomness: Uint8Array[] = [];
+        for (let i = 0; i < job.sets.length; i++) {
+          randomness.push(randomBytesNonZero(8));
+        }
+        return {
+          opts: job.opts,
+          sets: [
+            {
+              message: job.message,
+              publicKey: bls.PublicKey.aggregate(job.sets.map((set, i) => set.publicKey.multiplyBy(randomness[i]))),
+              signature: bls.Signature.aggregate(signatures.map((sig, i) => sig.multiplyBy(randomness[i]))),
```

Comment:
@matthewkeil  just to clarify, this `multiplyBy` in this line and the above does not mutate data at all, instead it creates new pubkey/signature objects?

Reply:
Yes, confirmed.  It creates a new key in the native layer by serializing and deserializing the point and returning a new key from the new point